notes.lua

```diff
@@ -0,0 +1,324 @@
+    return self.viewport_pos.x ~=  df.global.window_x or
+        self.viewport_pos.y ~=  df.global.window_y or
+        self.viewport_pos.z ~=  df.global.window_z
+end
+
+function NotesOverlay:onRenderFrame(dc)
+    if not df.global.pause_state and not dfhack.screen.inGraphicsMode() then
+        return
+    end
+
+    if self:viewportChanged() then
+        self:reloadVisibleNotes()
+    end
+
+    dc:map(true)
+
+    local texpos = dfhack.textures.getTexposByHandle(green_pin[1])
+    dc:pen({fg=COLOR_BLACK, bg=COLOR_LIGHTCYAN, tile=texpos})
+
+    for _, note in pairs(self.notes) do
+        dc
+            :seek(note.screen_pos.x, note.screen_pos.y)
+            :char('N')
+    end
+
+    dc:map(false)
+end
+
+function NotesOverlay:reloadVisibleNotes()
+    self.notes = {}
+
+    local viewport = guidm.Viewport.get()
+    self.viewport_pos = {
+        x=df.global.window_x,
+        y=df.global.window_y,
+        z=df.global.window_z
+    }
+
+    for _, map_point in ipairs(map_points) do
+        if viewport:isVisible(map_point.pos) then
+            local screen_pos = viewport:tileToScreen(map_point.pos)
+            table.insert(self.notes, {
+                point=map_point,
+                screen_pos=screen_pos
+            })
+        end
+    end
+end
+
+NoteManager = defclass(NoteManager, gui.ZScreen)
+NoteManager.ATTRS{
+    focus_path='notes/note-manager',
+    note=DEFAULT_NIL,
+    on_update=DEFAULT_NIL,
+}
+
+function NoteManager:init()
+    local edit_mode = self.note ~= nil
+
+    self:addviews{
+        widgets.Window{
+            frame={w=35,h=20},
+            frame_inset={t=1},
+            subviews={
+                widgets.HotkeyLabel {
+                    key='CUSTOM_ALT_N',
+                    label='Name',
+                    frame={t=0},
+                    on_activate=function() self.subviews.name:setFocus(true) end,
+                },
+                text_editor.TextEditor{
+                    view_id='name',
+                    frame={t=1,h=3},
+                    frame_style=gui.FRAME_INTERIOR,
+                    init_text=self.note and self.note.point.name or '',
+                    init_cursor=1
+                },
+                widgets.HotkeyLabel {
+                    key='CUSTOM_ALT_C',
+                    label='Comment',
+                    frame={t=5},
+                    on_activate=function() self.subviews.comment:setFocus(true) end,
+                },
+                text_editor.TextEditor{
+                    view_id='comment',
+                    frame={t=6,b=3},
+                    frame_style=gui.FRAME_INTERIOR,
+                    init_text=self.note and self.note.point.comment or '',
+                    init_cursor=1
+                },
+                widgets.Panel{
+                    view_id='buttons',
+                    frame={b=0,h=2},
+                    autoarrange_subviews=true,
+                    subviews={
+                        widgets.HotkeyLabel{
```

Comment:
the save/create and delete button would make better use of space if they were on the same line